buildspec/linuxUiTests.yml

```diff
@@ -21,6 +21,12 @@ phases:
     commands:
       - apt-get update
       - apt-get install -y xvfb icewm procps ffmpeg libswt-gtk-3-java msbuild
+      - mkdir -p /tmp/.aws
+      - aws sts assume-role --role-arn $ASSUME_ROLE_ARN --role-session-name ui-test > /tmp/.aws/creds.json
+      - export KEY_ID=`jq -r '.Credentials.AccessKeyId' /tmp/.aws/creds.json`
+      - export SECRET=`jq -r '.Credentials.SecretAccessKey' /tmp/.aws/creds.json`
+      - export TOKEN=`jq -r '.Credentials.SessionToken' /tmp/.aws/creds.json`
+      - echo -e "[default]\naws_access_key_id=$KEY_ID\naws_secret_access_key=$SECRET\naws_session_token=$TOKEN" > /tmp/.aws/credentials
```

Comment:
are KEY_ID/SECRET/TOKEN needed outside of this? If not, better not to use `export`, else (if codebuild works like travis) those env vars may show up in the environment of the main build, where because of their generic names they may confuse children, or show in logs, etc. Even if not, it's worth following good hygiene, to avoid copy-paste spread of these habits. 

Something like this:


```
- >
  KEY_ID="$(jq -r '.Credentials.AccessKeyId' /tmp/.aws/creds.json)"
  SECRET="$(jq -r '.Credentials.SecretAccessKey' /tmp/.aws/creds.json)"
  TOKEN="$(jq -r '.Credentials.SessionToken' /tmp/.aws/creds.json)"
  echo -e "[default]\naws_access_key_id=$KEY_ID\naws_secret_access_key=$SECRET\naws_session_token=$TOKEN" > /tmp/.aws/credentials
```

In that example, `export` isn't needed because the env vars are defined before the `echo` command (YAML `>` joins the lines to make a single space-separated line).

- btw: prefer `$(..)` instead of backticks for readability
- `$(..)` (or backticks) should be quoted

Reply:
thanks!